conanfile.py

```diff
@@ -49,13 +48,34 @@ def generate(self):
         tc.generate()
 
     def build(self):
+        if os.environ.get("BUILD_DIR_FULL_PATH") is not None:
+            return
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = CMake(self)
-        cmake.install()
+        if os.environ.get("BUILD_DIR_FULL_PATH") is not None:
+            self.output.info("BUILD_DIR_FULL_PATH: " + os.environ.get("BUILD_DIR_FULL_PATH"))
+            build_folder_path = os.environ.get("BUILD_DIR_FULL_PATH")
+            self.output.info("BUILD_DIR_FULL_PATH: " + build_folder_path)
+            self.copy(pattern='*', dst='.', src=('%s/install') % (build_folder_path), symlinks=True, ignore_case=False)
+        else:
+            cmake = CMake(self)
+            cmake.install()
 
     def package_info(self):
-        self.cpp_info.libs = ["up-cpp"]
+        if os.environ.get("BUILD_DIR_FULL_PATH") is not None:
+            self.cpp_info.set_property("cmake_file_name", "up-cpp")
+            self.cpp_info.set_property("cmake_target_name", "up-cpp::up-cpp")
+            self.cpp_info.set_property("pkg_config_name", "up-cpp")
+            self.cpp_info.libs = collect_libs(self)
+
+            self.cpp_info.set_property("cmake_target_name", "up-cpp::up-cpp")
+            self.cpp_info.requires = ["spdlog::spdlog", "protobuf::protobuf"]
+
+            self.cpp_info.names["cmake_find_package"] = "up-cpp"
+            self.cpp_info.names["cmake_find_package_multi"] = "up-cpp"
+
+        else:
+            self.cpp_info.libs = ["up-cpp"]
```

Comment:
These are fairly significant changes. Can you provide an explanation of what these changes are and what they do?